client/src/NetworkClient.ts

```diff
@@ -58,6 +71,17 @@ export class NetworkClient {
     private readonly _endpoint: string;
     private _eventClient!: EventClient;
     private $iframe!: HTMLIFrameElement;
+    private _apiLoadingState: 'not-started' | 'ready' | 'failed' = 'not-started';
+    private _consensus: 'syncing' | 'established' | 'lost' = 'syncing';
+    private _peerCount: number = 0;
+    private _headInfo: { height: number, globalHashrate: number } = { height: 0, globalHashrate: 0 };
+    private _balances: Map<string, number> = new Map<string, number>();
+    private _pendingTransactions: Map<string, Partial<DetailedPlainTransaction>>
+        = new Map<string, Partial<DetailedPlainTransaction>>();
+    private _expiredTransactions: Array<[number, string]> = [];
+    private _minedTransactions: Map<string, DetailedPlainTransaction> = new Map<string, DetailedPlainTransaction>();
+    private _relayedTransactions: Map<string, Partial<DetailedPlainTransaction>> =
+        new Map<string, Partial<DetailedPlainTransaction>>();
```

Comment:
I think instead of putting all these properties on the class directly, they could all go into a `private _cache = {...}` object.

Reply:
I think it's nicer the way it is

Reply:
My argument would be that it's cleaner, because it becomes obvious what a variable represents. Now you are just polluting the NetworkClient namespace (although with private members).

Reply:
But this way the type annotation and initial value can be seen directly where the variable is defined.
Otherwise it would be something bulky like:
```ts
private _cache: {
    consensusState: 'established' | 'lost' | 'syncing',
    peerCount: number,
} = {
    consensusState: 'syncing',
    peerCount: 0,
};
```

Or less bulky by defining a type, but then the type would not be annotated directly where the variables are defined:
```ts
private _cache: NetworkClientCache = {
    consensusState: 'syncing',
    peerCount: 0,
};
```